struct-patch-derive/src/lib.rs

```diff
@@ -100,6 +100,27 @@ impl Patch {
         #[cfg(not(feature = "status"))]
         let patch_status_impl = quote!();
 
+        let add_impl = quote! {
+            impl core::ops::Add<#name #generics> for #struct_name #generics #where_clause {
+                type Output = #struct_name;
+
+                fn add(mut self, rhs: #name #generics) -> Self {
+                    self.apply(rhs);
+                    self
+                }
+            }
+
+            impl core::ops::Add<#struct_name #generics> for #name #generics #where_clause {
+                type Output = #struct_name;
```

Comment:
Don't forget the generics:

```suggestion
                type Output = #struct_name #generics;
```